### PR TITLE
POC: use flagset

### PR DESCRIPTION
**Pull request by gclapperton**
A quick POC to illustrate how to use a custom FlagSet

### CODE REVIEW

File: jira_test.go [hunk 1]
```diff
@@ -777,38 +780,40 @@ func TestSetOptionMixFunc(t *testing.T) {
...
 	// reset command line arg
-	oldArgs := os.Args
-	defer func() { os.Args = oldArgs }()
-
-	os.Args = append(os.Args, "--token=123")
-	os.Args = append(os.Args, "--type=license")
-	os.Args = append(os.Args, "--assigneeId=654")
-	os.Args = append(os.Args, "--api=http://snyk.io/api")
-	os.Args = append(os.Args, "--configFile=./fixtures")
+	args := []string{
```
Comment (gclapperton) [inline jira_test.go:783]: Arguments as they would be if we called `os.Args[1:]` (the first arg is the name of the tool)

File: jira_test.go [hunk 2]
```diff
@@ -777,38 +780,40 @@ func TestSetOptionMixFunc(t *testing.T) {
...
+		"--api=http://snyk.io/api",
+		"--configFile=./fixtures",
+		"--orgID=456",
+		"--jiraProjectID=15699",
+	}
 
 	options := flags{}
-	options.setOption()
+	// consider explicitly passing CLI args into the fn, rather than relying on global variables (ie. os.Args)
+	options.setOption(args)
```
Comment (gclapperton) [inline jira_test.go:795]: 2. Pass the args explicitly

File: utils.go [hunk 3]
```diff
@@ -106,68 +106,74 @@ Function setOption
...
 	var debug bool
 	var dryRun bool
 	var configFilePtr *string
 	v := viper.New()
 
+	// pflag uses a default global, shared FlagSet stored in the var `pflag.CommandLine`.
+	// Each test will use this FlagSet unless we use create a one explicitly. I personally almost
+	// always try to avoid globals, except when registering code (ie. Cobra)
+	// https://github.com/spf13/pflag/blob/v1.0.5/flag.go#L1188
+	fs := pflag.NewFlagSet("test", pflag.ContinueOnError)
```
Comment (gclapperton) [inline utils.go:121]: 3. Create a custom FlagSet

File: jira_test.go [hunk 4]
```diff
@@ -777,38 +780,40 @@ func TestSetOptionMixFunc(t *testing.T) {
...
-		priorityIsSeverity:     true,
-		priorityScoreThreshold: 20,
-		projectID:              "",
-		severity:               "critical",
+		assigneeID: "654",
+		// 	assigneeName:           "",
+		// 	debug:                  false,
+		// 	dryRun:                 false,
+		issueType:      "license",
+		jiraTicketType: "Bug",
```
Comment (mathild3r) [inline jira_test.go:810]: I see that you changed the result here, but it should, this test is using both flags and config file and the jiraTicketType is set at Task in the config file, it should not be set to the default value, it the same with the severity.

**Pull request closed without merging**